src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers.Common/SyntaxNodeExtensions.cs

```diff
@@ -303,5 +303,120 @@ public static StringComparer GetStringComparer(this SyntaxNode? node)
                 LanguageNames.VisualBasic => StringComparer.OrdinalIgnoreCase,
                 _ => throw new NotImplementedException(Resources.UnknownLanguage),
             };
+
+        /// <summary>
+        /// Attempts to determine if a given syntax node corresponds to a type symbol
+        /// based on the node's usage by its parent. This is a best guess in cases where
+        /// semantic information is not available.
+        /// </summary>
+        /// <remarks>
+        /// Checking the symbol in the semantic model is preferred and this method
+        /// should only be used when no symbol information is available.
+        /// </remarks>
+        /// <param name="node">The syntax node to investigate.</param>
+        /// <returns>True if it appears the syntax node is used in a context where a type
+        /// symbol would usually make sense.</returns>
+        public static bool IsTypeSyntax(this SyntaxNode? node)
+        {
+            var parentNode = node?.Parent;
+
+            if (parentNode is null)
+            {
+                return false;
+            }
+
+            // Object creation
+            if (parentNode.IsKind(CS.SyntaxKind.ObjectCreationExpression) || parentNode.IsKind(VB.SyntaxKind.ObjectCreationExpression))
+            {
+                return true;
+            }
+
+            // Static member access
+            if ((parentNode is CSSyntax.MemberAccessExpressionSyntax csMAE && csMAE.Expression.IsEquivalentTo(node))
```

Comment:
I think we have some generalized syntax helpers that may simplify this pattern and allow for language agnostic

Reply:
Good point. We have `IsMemberAccessExpression` in SyntaxNodeExtensions.Predicates. I'm not sure if I want to use that here, though, because in addition to checking that the parent is an MAE, I'm also checking that the node is the parent's 'expression' (as opposed to its Name) because we want to return true for `Controller.Foo` but not for `Controller.Foo`.

We could add another extension method that looks for that sort of MAE format, but I'm not sure if it's general-purpose enough to be worth extracting it that way.